src/emulator/gdbstub/src/gdb.cpp

```diff
@@ -258,46 +293,74 @@ static std::string cmd_read_registers(HostState &state, PacketCommand &command)
     return stream.str();
 }
 
-static std::string cmd_write_registers(HostState &state, PacketCommand &command) { return ""; }
+static std::string cmd_write_registers(HostState &state, PacketCommand &command) {
+    if (state.gdb.current_thread == -1)
+        return "";
+    CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
+
+    const std::string content = content_string(command).substr(1);
+
+    for (uint32_t a = 0; a < content.size() / 8; a++) {
+        uint32_t value = parse_hex(content.substr(a * 8, 8));
+        modify_reg(cpu, a, value);
+    }
+
+    return "OK";
+}
 
 static std::string cmd_read_register(HostState &state, PacketCommand &command) {
     if (state.gdb.current_thread == -1)
         return "";
     CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
 
-    std::string content = content_string(command);
+    const std::string content = content_string(command);
     int32_t reg = parse_hex(content.substr(1, content.size() - 1));
 
     return be_hex(fetch_reg(cpu, static_cast<uint32_t>(reg)));
 }
 
 static std::string cmd_write_register(HostState &state, PacketCommand &command) {
-    return "";
+    if (state.gdb.current_thread == -1)
+        return "";
+    CPUState &cpu = *state.kernel.threads[state.gdb.current_thread]->cpu.get();
+
+    const std::string content = content_string(command);
+    uint32_t equal_index = content.find('=');
+    int32_t reg = parse_hex(content.substr(1, equal_index - 1));
+    uint32_t value = parse_hex(content.substr(equal_index + 1));
+    modify_reg(cpu, reg, value);
+
+    return "OK";
 }
 
-static std::string cmd_read_memory(HostState &state, PacketCommand &command) {
-    std::string content = content_string(command);
-    size_t pos = content.find(',');
+static bool check_memory_region(Address address, Address length, MemState &mem) {
+    const auto page_first = static_cast<uint32_t>(address / KB(4));
+    const auto page_length = static_cast<uint32_t>(length + KB(4)) / KB(4);
 
-    std::string first = content.substr(1, pos - 1);
-    std::string second = content.substr(pos + 1, content.size() - pos);
-    uint32_t address = parse_hex(first);
-    uint32_t length = parse_hex(second);
+    const auto pages_begin = mem.allocated_pages.begin();
+    const auto range_end = pages_begin + page_first + page_length;
+
+    const bool memory_none = std::find(pages_begin + page_first, range_end, 0) != range_end;
+    const bool memory_null = std::find(pages_begin + page_first, range_end, 1) != range_end;
+    const bool memory_safe = !(memory_none || memory_null);
```

Comment:
Really informative, however I don't know if I can take extra class tomorrow or i drop it since the wind is too large and when i go to sleep there might be a chance of rain so hard that i will sleep past where i have to go to class.

My grandfather is the leader of a secert guild that teach kong fu technique. One day he passed the technique on me and suddenly I am being chased by bad guys, so I have to transfer school a lot. I don't have many friends, but when I enter Inoba High School there is this girl named Kazuki who befriended me. We have fun a lot. but turns out she was an enemy too, she was stabbing me from the back. When I was being chased by here, i run and hit another girl named Mina who fell in love with me. Kazuki then stop chasing and being a love rival. But during that time, another girl steps in. Her name is Trish and she came with a bunch of weird Italians who seek me to help them find the treasure of a secert Italian gangster leader. From there, all things are messed up. What should I do

Reply:
Dude, that situation sounds tight. Hope you find a way out of it soon.

If the bad guys are still chasing you I would suggest talking to the Move Deleter in Lilycove City.

Reply:
I afraid I'm gonna get danger since Trish and the gang can't protect me all the time, and Kazuki will stop loving me because I'm not cool ass anymore, and so do Mina, she will try to kill me again.  This is really tight. 

Maybe I will get the treasure and go home first, than drive my helicopter to Lilycove city with gon from hunter x hunter, which is the show i never see but the name keeps popping up on my twitter.com page

Reply:
Why did i wrote this

Reply:
It was fun.